### PR TITLE
NEO-2327 // update Table's select / clear functionality

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Table story](https://deploy-preview-482--neo-react-library-storybook.netlify.app/?path=/story/components-table--editable-data)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] ~tagged `@avaya-dux/dux-devs`~

Updating the Table's select / clear functionality. See [Jira Issue](https://avaya.atlassian.net/browse/NEO-2327) for details.

`@avaya-dux/dux-design`
- review header and body select/clear wording
- review header and body select/clear functionality
- NOTE: I have added additional functionality for when there is a single page, see [Jira Issue](https://avaya.atlassian.net/browse/NEO-2327) for details

@avaya-dux/dux-devs: Design requested an update to functionality. Tagging you now that there has been a functional change.

EXAMPLE GIFS:

When there are multiple pages:
multiple pages example

When there is a single page:
single page example

## Summary by CodeRabbit

- **Style**
	- Minor formatting adjustment in the styling for the table component, ensuring adherence to style guidelines without affecting functionality or visual output.
	- Improved styling for table rows and buttons in dark mode, enhancing visibility and user experience.

- **New Features**
	- Enhanced selection logic for the table component, improving clarity and precision of selected row visibility on the current page.

### CODE REVIEW

**Comment by Justin-Avaya** [comment]
Issue 
The menu option, "Clear all items on all pages" should not appear when items are selected on only one of multiple pages.

In the following examples, items are selected but only on the current page (page 1):

Screenshot 2024-08-01 at 6 42 35 AM

Screenshot 2024-07-31 at 3 09 08 PM

**Comment by joe-s-avaya** [comment]
> Issue The menu option, "Clear all items on all pages" should not appear when items are selected on only one of multiple pages.

@Justin-Avaya + @matthewAvaya updated. Note that I have written the code such that if there are any items selected that are not on the current page, "Clear all items on all pages" will be presented. Example 👇

File: src/components/Table/TableComponents/TableHeader.tsx [hunk 1]
```diff
@@ -356,11 +356,11 @@ const TableSelectionCheckboxAndMenu: TableHeaderComponentType = ({
 
 	const ClearAllPages = useMemo(() => {
 		const hasMoreThanOnePage = page.length !== rows.length;
-		const someRowsAreSelected = rows.some(
-			(row) => selectedRowIds[row.id] && !row.original.disabled,
+		const someRowsAreSelectedThatAreNotOnCurrentPage = rows.some(
+			(row) => selectedRowIds[row.id] && !page.find((p) => p.id === row.id),
```
Comment (yangAtSpoken) [inline src/components/Table/TableComponents/TableHeader.tsx:360]: p is a bit misleading.  Please use r.

Comment (joe-s-avaya) [reply src/components/Table/TableComponents/TableHeader.tsx]: good call, updated

**Review by Justin-Avaya — APPROVED** [review]

**Comment by matthewAvaya** [comment]
Change text link inside blue bar to be blue/600 so it passes contrast ratio.

**Comment by joe-s-avaya** [comment]
> Change text link inside blue bar to be blue/600 so it passes contrast ratio.

@matthewAvaya updated

**Review by matthewAvaya — APPROVED** [review]

**Pull request merged**